core/src/main/java/org/mule/runtime/core/message/AttachmentAttributes.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.message;
+
+import static java.util.Objects.requireNonNull;
+
+import com.google.common.collect.ImmutableMap;
+
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.Map;
+
+/**
+ * Representation of metadata associated to an attachment.
+ *
+ * @since 4.0
+ */
+public class AttachmentAttributes extends BaseAttributes
```

Comment:
Shouldn't this be PartAttributes?  Can this be an inner class in MultiPartPayload maybe?